src/App.tsx

```diff
@@ -15,36 +16,94 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  alphabet = 'alphabet',
+  length = 'length',
+  default = 0,
+}
+
+// type SortConfig = {
+//   type: string | null;
+//   reversed: boolean;
+// };
+
+export const App = () => {
+  const [sortType, setSortType] = useState('');
+  const [reversed, setReversed] = useState('');
+
+  function getSortedList(
+    goods: string[],
+    sortBy: string,
+    isReverse: string | boolean,
+  ) {
+    const sortedList = [...goods].sort((good1, good2) => {
+      switch (sortBy) {
+        case SortType.alphabet:
+          return good1.localeCompare(good2);
+        case SortType.length:
+          return good1.length - good2.length;
+        default:
+          return SortType.default;
+      }
+    });
+
+    return isReverse ? sortedList.reverse() : sortedList;
+  }
+
+  const reset = () => {
+    setReversed('');
+    setSortType('');
+  };
+
+  const visibleGoods = getSortedList(goodsFromServer, sortType, reversed);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          onClick={() => setSortType(SortType.alphabet)}
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sortType !== SortType.alphabet,
+          })}
+        >
           Sort alphabetically
         </button>
-
-        <button type="button" className="button is-success is-light">
+        <button
+          onClick={() => setSortType(SortType.length)}
+          type="button"
+          className={classNames('button', 'is-succes', {
+            'is-light': sortType !== SortType.length,
+          })}
+        >
           Sort by length
         </button>
-
-        <button type="button" className="button is-warning is-light">
+        <button
+          onClick={() => setReversed(!reversed)}
```

Comment:
The `onClick` handler for the Reverse button directly sets the `reversed` state without a verb in its name. Consider renaming it to `handleReverseClick` or similar for better readability and consistency with the [CODE STYLE] checklist.